src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+        private readonly object _lock;
+        private readonly bool _shouldDispose;
+        private TimeSpan _newLineSleepTime;
+        private string _romType;
+        private Encoding _characterEncoding;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="LcdConsole"/> class using the specified LCD low-level interface. 
+        /// This class automatically configures the low-level interface. Do not use the low-level interface at the same time. 
+        /// </summary>
+        /// <param name="lcd">The low-level LCD interface.</param>
+        /// <param name="romType">Name of character ROM of display. Currently supported types: A00 and A02.</param>
+        /// <param name="shouldDispose">If the class should dispose the LCD driver when it is disposed. Defaults to true</param>
+        public LcdConsole(ICharacterLcd lcd, String romType, bool shouldDispose = true)
+        {
+            _lcd = lcd;
+            _shouldDispose = shouldDispose;
+            _romType = romType;
+            _characterEncoding = null;
+            Size = lcd.Size;
+            _currentData = new StringBuilder[Size.Height];
+            CursorLeft = 0;
+            CursorTop = 0;
+            NewLineSleepTime = TimeSpan.Zero;
+            _lock = new object();
+            _lcd.UnderlineCursorVisible = false;
+            _lcd.BlinkingCursorVisible = false;
+            _lcd.DisplayOn = true;
+            _lcd.BacklightOn = true;
+            ClearStringBuffer();
+            _lcd.Clear();
+        }
+
+        /// <summary>
+        /// Position of the cursor, from left.
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorLeft
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Position of the cursor, from top
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorTop
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// If larger than Zero, an extra wait is introduced each time a newline is written. Can be used to render entire paragraphs to the display, 
+        /// but keeping it readable. 
+        /// </summary>
+        public TimeSpan NewLineSleepTime
+        {
+            get
+            {
+                return _newLineSleepTime;
+            }
+
+            set
+            {
+                if (value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), "Timespan must be positive");
+                }
+                _newLineSleepTime = value;
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables the backlight
+        /// </summary>
+        public bool BacklightOn
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _lcd.BacklightOn;
```

Comment:
does reading need a lock?

Reply:
Basically correct, but we shouldn't make any assumptions on the underlying interface here. This might as well trigger some device access.

Reply:
fair enough